core/src/identity/rsa.rs

```diff
@@ -60,6 +60,16 @@ impl Keypair {
             Err(e) => Err(SigningError::new("RSA", e))
         }
     }
+
+    /// Sign a message with this keypair using a custom digest algorithm.
+    pub fn sign_custom(&self, data: &[u8], algorithm: &'static dyn RsaEncoding) -> Result<Vec<u8>, SigningError> {
```

Comment:
Let's not expose the types of external libraries as part of our API, especially not the ones of `ring`. We want to be able to change implementation details without breaking the API.

Reply:
Sure, would you prefer to remove this all-together or have libp2p maintain it's own set of allowed algorithms?

Reply:
I would say it depends on your use-case. I see the `identity.rs` module as some utilities that are used in common by libp2p-noise and libp2p-secio. However I don't think libp2p-core should turn into a cryptographic library.

Reply:
Yeah, I've definitely run into this problem. I'm fine with leaving out some of these functions (I don't need them at the moment, maybe someone else might later). 

I'm building a new protocol, which originally, I thought would be nice to interop with current libp2p keys so that a singular key could work for the TLS/Secio layer and also at the protocol layer. In particular it uses diffie-hellman on the keypairs, which is quite difficult for RSA and Ed25519. I liked the key-abstraction but at this point, it's not worth building the implementation for all libp2p key types.